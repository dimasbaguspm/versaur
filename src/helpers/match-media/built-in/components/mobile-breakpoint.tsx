import { BREAKPOINT_MOBILE } from '../../constants'
import { MatchMedia } from '../../match-media'

/**
 * Props for breakpoint components
 */
interface BreakpointComponentProps {
  children: React.ReactNode
}

/**
 * Component that only renders children on mobile viewports
 *
 * @example
 * ```tsx
 * <MobileBreakpoint>
 *   <MobileOnlyContent />
 * </MobileBreakpoint>
 * ```
 */
export function MobileBreakpoint({ children }: BreakpointComponentProps) {
  return <MatchMedia query={BREAKPOINT_MOBILE}>{children}</MatchMedia>
}
