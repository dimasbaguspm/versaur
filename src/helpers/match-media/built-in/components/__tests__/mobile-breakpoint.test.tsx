import { render, screen } from '@testing-library/react'
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest'
import { MobileBreakpoint } from '../mobile-breakpoint'
import {
  createMatchMediaMock,
  setupMatchMediaMock,
} from '../../../test-helpers'

describe('MobileBreakpoint', () => {
  const { beforeEach: setupBefore, afterEach: setupAfter } =
    setupMatchMediaMock()

  beforeEach(() => {
    setupBefore()
  })

  afterEach(() => {
    setupAfter()
  })

  it('should render children when viewport is mobile', () => {
    const mockMatchMedia = createMatchMediaMock(true)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    render(
      <MobileBreakpoint>
        <div data-testid='mobile-content'>Mobile Content</div>
      </MobileBreakpoint>
    )

    expect(screen.getByTestId('mobile-content')).toBeInTheDocument()
  })

  it('should not render children when viewport is not mobile', () => {
    const mockMatchMedia = createMatchMediaMock(false)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    render(
      <MobileBreakpoint>
        <div data-testid='mobile-content'>Mobile Content</div>
      </MobileBreakpoint>
    )

    expect(screen.queryByTestId('mobile-content')).not.toBeInTheDocument()
  })

  it('should match snapshot when rendered', () => {
    const mockMatchMedia = createMatchMediaMock(true)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    const { asFragment } = render(
      <MobileBreakpoint>
        <div>Mobile Content</div>
      </MobileBreakpoint>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should not affect accessibility tree when hidden', () => {
    const mockMatchMedia = createMatchMediaMock(false)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    const { container } = render(
      <MobileBreakpoint>
        <div role='navigation'>Mobile Navigation</div>
      </MobileBreakpoint>
    )

    expect(
      container.querySelector('[role="navigation"]')
    ).not.toBeInTheDocument()
  })

  it('should preserve accessibility attributes when shown', () => {
    const mockMatchMedia = createMatchMediaMock(true)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    render(
      <MobileBreakpoint>
        <nav role='navigation' aria-label='Mobile Navigation'>
          <button>Menu</button>
        </nav>
      </MobileBreakpoint>
    )

    const nav = screen.getByRole('navigation', { name: 'Mobile Navigation' })
    expect(nav).toBeInTheDocument()
  })
})
