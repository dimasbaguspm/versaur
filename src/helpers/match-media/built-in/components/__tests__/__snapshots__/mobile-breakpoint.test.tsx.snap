// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MobileBreakpoint > should match snapshot when rendered 1`] = `
<DocumentFragment>
  <div>
    Mobile Content
  </div>
</DocumentFragment>
`;
